The reason this won't affect glibc is that the default locale is `C`
so this is a no-op, but on musl the default locale is `C.UTF-8`.

The actual test failure is some `grep` is piped some binary that is
not valid utf8. This can be seen in glibc with:

    $ echo -e 'A\xa4' | LC_ALL=C.UTF-8 grep A
    grep: (standard input): binary file matches

--- a/src/tests/t.frame	2019-10-10 17:04:33.000000000 -0300
+++ b/src/tests/t.frame	2024-06-10 22:46:07.546991269 -0300
@@ -88,6 +88,9 @@
 unset LANG || :
 unset LC_COLLATE || :
 unset LC_ALL || :
+# we set LC_ALL=C because this is what is really wanted; on musl libc,
+# the default is C.UTF-8 which breaks t.dump.load and t.nonascii.replace
+export LC_ALL=C
 i=0
 INSERT=.
 while [ $i -lt 3 ] ; do
